### ISSUE TITLE
[1.16.x] Hostile mobs appearing invisible and invulnerable.

### ISSUE THREAD

**Issue by dabulue** [sp614x/optifine#5016]
https://github.com/sp614x/optifine/issues/4351

This exact issue is still in this mod and it makes the game unplayable with OptiFine. Stop closing issues that are not fixed.

**Comment by Lythael** [comment]
Yep its not fixed.
It happens in 1.16.4 release and in 1.16.3 aswell (as i tried to downgrade to the version that supposedly "fixed" this)
So far seen Piglins , Skeletons , Drowned going invisible untill i exit to main menu and re-enter.

This happens in vanilla game (no mods or shaders installed just Optifine)

**Comment by legendaryWalter** [comment]
Happens in previous versions as well such as 1.16.1 and 1.15.2. It is very annoying. I can't even raid a bastion because of some invisible Piglin Brutes.

**Comment by Lythael** [comment]
> Happens in previous versions as well such as 1.16.1 and 1.15.2. It is very annoying. I can't even raid a bastion because of some invisible Piglin Brutes.

Yea ;/ for the time being if you don't want to uninstall Optifine. Exit to menu and re-enter when you are near Bastion or transitioned between dimensions. Cause idk if consistent but i noticed this invisiblity bug happens after i travel between dimensions then there is a very high chance afterwards that i will encounter invisible mobs.

**Comment by VirtualViolet** [comment]
I've had the same issue. No other mods installed, just Optifine. Ran into an illager squad and only 2 of them were visible, at least one more was invisible (I ran from it and eventually ran into a creeper that exploded and killed it, lol). Also, there were a couple of cats in a village that were invisible for the longest time, which I eventually got to reappear somehow. I'm not sure what did it because I didn't know exactly where they were because they were running from me (I could hear them), but I disabled a few of the optimization settings before the first one reappeared. Now that I've chased it down and befriended it, it doesn't seem to be disappearing anymore. (Note: I have not yet changed dimensions on this world. Everything has been in the overworld.)

I'll update this if I get any more concrete information.

**Comment by Lythael** [comment]
> I've had the same issue. No other mods installed, just Optifine. Ran into an illager squad and only 2 of them were visible, at least one more was invisible (I ran from it and eventually ran into a creeper that exploded and killed it, lol). Also, there were a couple of cats in a village that were invisible for the longest time, which I eventually got to reappear somehow. I'm not sure what did it because I didn't know exactly where they were because they were running from me (I could hear them), but I disabled a few of the optimization settings before the first one reappeared. Now that I've chased it down and befriended it, it doesn't seem to be disappearing anymore. (Note: I have not yet changed dimensions on this world. Everything has been in the overworld.)
> 
> I'll update this if I get any more concrete information.

As i read more and more through older posts about this issue i think it might be related to Dynamic Lighting setting.
Having it off fixes the issue (apparently) but i am not sure i'll do a playthrough with it off and i will report if i notice invisible mobs again.

**Comment by dabulue** [comment]
> > I've had the same issue. No other mods installed, just Optifine. Ran into an illager squad and only 2 of them were visible, at least one more was invisible (I ran from it and eventually ran into a creeper that exploded and killed it, lol). Also, there were a couple of cats in a village that were invisible for the longest time, which I eventually got to reappear somehow. I'm not sure what did it because I didn't know exactly where they were because they were running from me (I could hear them), but I disabled a few of the optimization settings before the first one reappeared. Now that I've chased it down and befriended it, it doesn't seem to be disappearing anymore. (Note: I have not yet changed dimensions on this world. Everything has been in the overworld.)
> > I'll update this if I get any more concrete information.
> 
> As i read more and more through older posts about this issue i think it might be related to Dynamic Lighting setting.
> Having it off fixes the issue (apparently) but i am not sure i'll do a playthrough with it off and i will report if i notice invisible mobs again.

Turning off dynamic lights doesn't fix it. I tried. It doesn't work.

**Comment by Lythael** [comment]
> > > I've had the same issue. No other mods installed, just Optifine. Ran into an illager squad and only 2 of them were visible, at least one more was invisible (I ran from it and eventually ran into a creeper that exploded and killed it, lol). Also, there were a couple of cats in a village that were invisible for the longest time, which I eventually got to reappear somehow. I'm not sure what did it because I didn't know exactly where they were because they were running from me (I could hear them), but I disabled a few of the optimization settings before the first one reappeared. Now that I've chased it down and befriended it, it doesn't seem to be disappearing anymore. (Note: I have not yet changed dimensions on this world. Everything has been in the overworld.)
> > > I'll update this if I get any more concrete information.
> > 
> > 
> > As i read more and more through older posts about this issue i think it might be related to Dynamic Lighting setting.
> > Having it off fixes the issue (apparently) but i am not sure i'll do a playthrough with it off and i will report if i notice invisible mobs again.
> 
> Turning off dynamic lights doesn't fix it. I tried. It doesn't work.

Oh .. bummer :(

**Comment by Javrel** [comment]
Im having this in 16.4, making my game unplayable I spent so long on a village to have a raid with invis witches n illagers screw it up

**Comment by Siuolthepic** [comment]
Alright.
Most likely this is a duplicate of #1892 . Do you guys mind moving the info you gathered about this over there? If any of you guys have a way to reproduce this the mod creator, Sp614x. Would appriciate it.
Ill wait for a contirbutor to judge if this is a duplicate or not though.
Also, when you guys are there you might wanna put in your specs.

**Comment by BobBobilou** [comment]
> Yep its not fixed.
> It happens in 1.16.4 release and in 1.16.3 aswell (as i tried to downgrade to the version that supposedly "fixed" this)
> So far seen Piglins , Skeletons , Drowned going invisible untill i exit to main menu and re-enter.
> 
> This happens in vanilla game (no mods or shaders installed just Optifine)
> 
> EDIT: On a side note they can still be attacked and killed tho its very tricky so just restarting is better.
> But yea this should be fixed (as in actually fixed not just make a placebo claim that its fixed) cause its no small issue and its game breaking for hardcore play getting killed by something you can't even see or attack (usually)
> 
> EDIT 2: Ok this is getting ridiculous ... Invisible Creeper destroyed my crop farm today.

I have download optifine for the past 6 month and each new version that I try since had this bug. I play hardcore so this bug is really breaking the game. This is an huge issus and it really need to be fix.
Also, I found out that if you use an sword and you punch an mob next to the invisible one you can hit it. Also, you can put it in fire by lighting the block below it. Please fix it this has made multiple month and not patch yet. 
Thanks

**Comment by tallcoleman** [comment]
Had this issue in 1.16.4 with a woodland mansion. As others have suggested, turning off dynamic lighting fixed it, and I was able to see many of the vindicators again.

Thought I was losing it taking damage to enemies I couldn't see...

**Comment by NicoTheCinderace** [comment]
I'm having the same issues as well, with occasional invisible Creepers and Zombies.

What's even causing this...?

**Comment by dpill83** [comment]
I created a server running Drehmal: PRIMORDIAL map and Optifine is a requirement to use it. I'm getting complaints that some mobs are invisible. Advised users to close the game and relaunch it and the mobs appear.

**Comment by Lythael** [comment]
> I created a server running Drehmal: PRIMORDIAL map and Optifine is a requirement to use it. I'm getting complaints that some mobs are invisible. Advised users to close the game and relaunch it and the mobs appear.

Yea tho you don't need to close the game just leave to main menu and re-enter the world it fixes that too.
But same as exiting the game this method just temporarily fixes the issue.

**Comment by lolcuslol** [comment]
I have this same issue with creepers and husks but i ve also seen drowned turn invisible and im playing on 1.16.4

**Comment by Teasy** [comment]
Thought I'd take this new Minecraft version for a spin, but shaders (thank you Optifine creators!) are essential to make the experience more life-like!
Within the first 15-30 minutes of getting wood, a shelter, some iron, and some crops, an **invisible drowned with trident** sneaks up on me in clear daylight (i'm sure he followed me after i went fishing) and before I realized what's going on, I'm dead! (2 hits!) xD
And this invisible drowned is now, alongside various creepers and skellies continuously spawnkilling me ... xD
I asked for a challenge, but I didn't expect the computer players to cheat! :P

**Comment by MavG8ER** [comment]
holy crap is this stupid bug still out i cant play very well without optifine like why have they not fixed it yet i had two bastions in a row with invisible mobs IM ACTUALLY GOING MENTAL CAUSE OF THIS

**Comment by MavG8ER** [comment]
6 MONTHS SINCE FIRST 1.16.4 OPTIFINE please for the love of god fix this bug

**Comment by sp614x** [comment]
Another round of testing triggered by #5183.
Tested with identical settings, same version, same shaders, new world - no luck, all mobs visible.
Made the drowned spawn particles in a wide radius to see them even if unvisible, wandered in the ocean for half an hour - all drowned still visible.
This may be hardware related, multi-core CPUs, cache synchronization or something similar.
The #5183 is for 12x Ryzen 5 3600.

**Comment by Lythael** [comment]
> Another round of testing triggered by #5183.
> Tested with identical settings, same version, same shaders, new world - no luck, all mobs visible.
> Made the drowned spawn particles in a wide radius to see them even if unvisible, wandered in the ocean for half an hour - all drowned still visible.
> This may be hardware related, multi-core CPUs, cache synchronization or something similar.
> The #5183 is for 12x Ryzen 5 3600.

I have a AMD A8-6600k it happens occassionally.
Very very rarely when just exploring Overworld.
But the chance of this bug to happen increases after i transition in or out of the Nether.
Bastion area is very prone to this invisibility glitch too.

**Comment by sp614x** [comment]
Here are 4 test versions based on pre10, with different changes that can help to reproduce or fix the bug.

Pre10a: 
 - all mobs emit particles to make them easy to spot
 - in spectator mode the terrain is not rendered which makes it easy to see all mobs
 - otherwise it is identical to pre10 and the bug should be easy to reproduce

https://optifine.net/tmp/ver/OptiFine_1.16.4_HD_U_G6_pre10a.jar

Pre10b, pre10c and pre10d 
- each one has a different entity optimization disabled
- based on pre10a (mobs emit particles, ...)

https://optifine.net/tmp/ver/OptiFine_1.16.4_HD_U_G6_pre10b.jar
https://optifine.net/tmp/ver/OptiFine_1.16.4_HD_U_G6_pre10c.jar
https://optifine.net/tmp/ver/OptiFine_1.16.4_HD_U_G6_pre10d.jar

Start with pre10a and try to reproduce the bug. All mobs emit particles, so seeing particles without a mob = invisible mob.
Changing to spectator mode will hide the terrain which helps to see all mobs in the nether.

When the bug is reliably reproducible, you can test with pre10b, pre10c and pre10d to see if the bug still exists.

**Comment by Lythael** [comment]
I'm having troubles installing these test versions.
They don't show up in the Launcher.

**Comment by Lythael** [comment]
> Here are 4 test versions based on pre10, with different changes that can help to reproduce or fix the bug.
> 
> Pre10a:
> 
> * all mobs emit particles to make them easy to spot
> * in spectator mode the terrain is not rendered which makes it easy to see all mobs
> * otherwise it is identical to pre10 and the bug should be easy to reproduce
> 
> https://optifine.net/tmp/ver/OptiFine_1.16.4_HD_U_G6_pre10a.jar
> 
> Pre10b, pre10c and pre10d
> 
> * each one has a different entity optimization disabled
> * based on pre10a (mobs emit particles, ...)
> 
> https://optifine.net/tmp/ver/OptiFine_1.16.4_HD_U_G6_pre10b.jar
> https://optifine.net/tmp/ver/OptiFine_1.16.4_HD_U_G6_pre10c.jar
> https://optifine.net/tmp/ver/OptiFine_1.16.4_HD_U_G6_pre10d.jar
> 
> Start with pre10a and try to reproduce the bug. All mobs emit particles, so seeing particles without a mob = invisible mob.
> Changing to spectator mode will hide the terrain which helps to see all mobs in the nether.
> 
> When the bug is reliably reproducible, you can test with pre10b, pre10c and pre10d to see if the bug still exists.

Soo I have been at it for an hour. I cannot reproduce the bug in:
OptiFine_1.16.4_HD_U_G6_pre10a
I've been in and out of the Nether testing this with both shaders on and off, near bastion and in overworld after leaving nether etc.

It just refuses to happen on demand even in the hot spots when bug tends to happen more often than usually.
Knowing my luck i wouldn't be surprised if i ran into it randomly while casually playing and frequency of this bug happening near bastion was just pure coincidence.

This bug happens rarely at random it seems :/
I'll try testing some more i'll report back if it happens again.

**Comment by Ruchian** [comment]
Minecraft 1.16.4
Optifine version: OptiFine HD U G5
Modpack: All the mods 6
Dedicated server.

I encountered this problem several times.
Invisible creeper, invisible drowned, invisible zombie
Invisible Apotheosis boss mob....

Invisible mobs are not just invisible, they are not there for the player. They're there on the server side though and therefore blocks cannot be placed where they are
They can however be suffocated. Sound events of course still play, but yeah... they are unselectable, untargetable, unhitable.

![2021-01-06_08 08 18](https://user-images.githubusercontent.com/24900404/103822656-d4b31f00-5070-11eb-8473-d1629d4f0671.png)

Here is an invisible zombie I managed to trap in a hole... 

The problem seems to occur more frequently when you travel a lot and when you have been online for a long time. (4+ hours)
And yes I no-life. I am easily playing for over 8 hours. xd
anyway, uh... this is very problematic.
I realized it was Optifine causing this because it affects villagers as well, even when they are sleeping. Some just won't load in on the client, even when they are close by.

I am hoping you could fix this at some point because, its kinda risky in game.

**Comment by sp614x** [comment]
Another report also showed that invisible mobs don't exist for the client at all. 
New version pre10e - the mob particles are generated on the server thread and added on the client thread.
https://optifine.net/tmp/ver/OptiFine_1.16.4_HD_U_G6_pre10e.jar
The particles should appear even if the mob doesn't exist for the client.
Mob particles without mob => invisible mob.

**Comment by sp614x** [comment]
Found one invisible mob

![image](https://user-images.githubusercontent.com/16635441/103925108-55c4f180-5117-11eb-8f54-5330722812d1.png)

**Comment by Ruchian** [comment]
sp614x, do you recommend me with this to take this hidden beta version and test it out? xd

I also wanted to note that the problem, once it starts becomes pretty frequent. I think it is related to them spawning, not just unloading. It is reproducable, but not in an exact manner of steps. The only hint I have is, chunk distance, (visibility)
I noticed Optifine tries to unload entities in far away chunks and reloads them in when close. Sometimes this happens very, rediculously late, as in you need to stand on top of a bed before you see the villager laying in it suddenly appear.
The other hint is not known entities (new ones,
Then lastly, you need to have overloaded some kind of list of entities.
It doesn't matter how many entities in a single place there were, although you will notice it sooner. All you need to have done is having seen many, many entities.

And yes I think without noticing some floating drops for example, items too become invisible. But the player will still pick these up, because the server handles this.

In other words, it could very well be the only mob around, and the only entity around the player, it is invisible, invulnerable, untargetable, etc.

You will encounter some of these hidden mobs at some point, its a dedicated server issue only, I am guessing since on a client server, I guess if it happens then. it never happened noticably for the player.

Anyway, that is all I have to share. Please test this, to confirm. xd

**Comment by DungInACup** [comment]
Just died to an invisible piglin while exploring a bastion in hardcore while using 1.16.4 OptiFine HD U G6 pre10.

**Comment by Sarkos** [comment]
This happens a lot in the Drehmal custom map, which has locations guarded by groups of illagers that do not despawn. Very often, 1 or 2 illagers out of the group will be invisible and untargetable.

Since other people have mentioned this bug occurring with piglins in bastions (which are pre-generated, non-despawning mobs), it seems likely that the bug only occurs with mobs that already existed in the chunk when the player arrived.

**Comment by Lythael** [comment]
> This happens a lot in the Drehmal custom map, which has locations guarded by groups of illagers that do not despawn. Very often, 1 or 2 illagers out of the group will be invisible and untargetable.
> 
> Since other people have mentioned this bug occurring with piglins in bastions (which are pre-generated, non-despawning mobs), it seems likely that the bug only occurs with mobs that already existed in the chunk when the player arrived.

Not only cause i've seen random spawning Creepers go invisible too but not as common as pre-generated piglins.

**Comment by KakashoLin** [comment]
Okay, I am able to effectively repeat this multiple times.  I noticed it happening again to my pet slime that I named yesterday. Two main things I did yesterday was going to the nether and working on a base there and going to a faraway biome for foxes. Testing today, going to the nether and back did not trigger the bug, but going to the far away biome, many chunks away triggered the bug. To make sure it was caused by this, I repeated the experiment three times, and each time the slime vanished but I could still hear it. And between each test, I saved and exited and reentered and the slime would always reappear. I attached the distance I traveled and the f3 report in case that shows something of interest. 

*Note: I also updated ticket 5196 with this information as well and other information leading to this discovery.


![2021-01-10_23 44 35](https://user-images.githubusercontent.com/77138726/104151035-85872a00-53a1-11eb-94ba-da7e596439ad.png)

![2021-01-10_23 51 44](https://user-images.githubusercontent.com/77138726/104151049-8fa92880-53a1-11eb-8356-96611d1a56a3.png)

**Comment by xFusakeruna** [comment]
This needs to be fixed!
I had full prot IV and pretty much instadied losing 60 hours of progress `-´ 
https://medal.tv/clips/41430733/d1337VF3qLtH

**Comment by KakashoLin** [comment]
Checked and the issue is still present in the latest update that was released a few days ago, and the bug is still repeatable.

**Comment by Lythael** [comment]
Found several invisible Drowned while exploring today:
https://i.imgur.com/JvJluyu.png
One was throwing trident at me.

Only way to attack him was with ranged weapons.
This was on the 1.16.5 Optifine release.

I hope i live to see the day that this will be finally fixed once and for all

**Comment by SkyeMors** [comment]
I've encountered this bug with a few (invisible) Pillagers that were wandering around one of their towers while I was exploring, a world reload fixed them.

Not sure if related, but it appears tile entities also have a slight transparency issue, most notably cauldrons. It was mentioned in this thread or a sister thread that any texture with slight transparency or full transparency causes the entire texture to be just a hair transparent alongside the transparent part.
I'll mess with my resource packs, as one of them adds a transparent portion to cauldrons, and edit with my results.

**Comment by Frappyy** [comment]
Wasn't aware of this issue until now when I lost my hardcore world to it. Didn't think to reload my world or anything either, this really fucking sucks.

**Comment by Lythael** [comment]
Still happens in OptiFine HD U G7 pre5
So i see you had no luck in figuring out how to fix this?

On a side note. Drowned near sunken ruins seems to be the most reliable way to reproduce this bug. It happens almost all the time near those and either i'm crazy or it happens way more often than in previous versions near those ruins.

**Comment by joseywoermann** [comment]
I have _never_ encountered this issue.

- Ryzen 5 2500U (4C/8T)
- Vega 8 integrated graphics
- 8gb RAM
- every Windows 10 version from 1903 - 20H2
- not sure what OF versions I used, but I know for sure that I played (almost) every version from 1.14.2 - 1.16.4 & 1.8.9

**Comment by PeteCozens** [comment]
Glad to hear that you're looking into this - thanks very much!

I get the occasional invisible spider (only red glow from eyes/mouth visible) and zombie (completely invisible) on 1.16.5-Optifine_HD_U_G6. I've no idea what activity might be causing them - there's no pattern I can identify as to when or why I encounter them.

PC Specs: AMD Ryzen 5 3600X 6-Core, 16GB RAM, Windows 10 Home, AMD Radeon RX 5500XT Graphics card. O/S and drivers all up-to-date. I've assigned 6GB RAM to Minecraft in the launcher, but this also happened when it was set to the default 2GB as well.

Options files as follows:
[options.txt](https://github.com/sp614x/optifine/files/5918892/options.txt)
[optionsof.txt](https://github.com/sp614x/optifine/files/5918893/optionsof.txt)

**Comment by SeaOfPixels** [comment]
Game breaking, should be at the top of the priority list

**Comment by Oakstar519** [comment]
I've gotten this bug a few times. I think the only thing in common between the mobs that went invisible was that they were persistent and it happened after loading/unloading chunks: the most recent two times were a named zombie and some pillagers from a raid.

PC specs: Intel i7 930 CPU, 24 GB RAM, Radeon R7 260X GPU, Windows 7 Ultimate. All drivers except the graphics driver are up to date, but updating the graphics driver doesn't fix the issue (and breaks other stuff.) Minecraft has 4GB of RAM allocated, but it's also happened with 2GB.

I've gotten this bug with normal OptiFine (at least once in 1.15.x and more than once in 1.16.x) as well as a custom Fabric pack with OptiFabric and a few vanilla-compatible mods. (I can provide a list, but my configuration changes so often that it probably wouldn't be accurate.)

[optionsof.txt](https://github.com/sp614x/optifine/files/5921082/optionsof.txt)
[options.txt](https://github.com/sp614x/optifine/files/5921083/options.txt)

**Comment by Lythael** [comment]
> I get the occasional invisible spider (only red glow from eyes/mouth visible)

That's actually not a bug its intentional.
Spiders sometimes inheret potion effects including invisibility.
You can tell this is not bug cause these spiders display a hitbox and can be attacked by melee.

While actual glitched invisible mobs are completely invisible including effects they had on them and they don't display a hitbox and cannot be damaged by melee (only ranged works)

**Comment by harmalom** [comment]
In my case it usually happens with the Pillagers, especially with the Patrols, sometimes 1 or 2, rarely 3 of the group, others that are usually invisible are the zombies, but these are usually rarer.

I use 
Shaders 
ComplementaryShaders_3.10

Resource Pack
Stoo's Deluxe Ressource Pack v1.10a
ComplementaryResources 3.10
VanillaTweaks_r999741

Mods List
https://pastebin.com/GGUw20Gd

**Comment by JackDotJS** [comment]
> Game breaking, should be at the top of the priority list

this is why the issue is pinned on this repo

**Comment by SeaOfPixels** [comment]
Seems to be maybe be an issue with mobs that spawn in with structures. This also sometimes happens when you relog into a world.

**Comment by SeaOfPixels** [comment]
Also they're not just invisible, they're invincible to melee attacks. This is not just a visual glitch.

**Comment by Oakstar519** [comment]
> Seems to be maybe be an issue with mobs that spawn in with structures. This also sometimes happens when you relog into a world.

In my experience it's persistent mobs, which includes structure-spawned mobs.

**Comment by dubif** [comment]
It appears that this bug is related to the "smooth world" perfomance option, turning it ON makes the zombie in my iron farm disappear, if I turn it OFF and get far away from the farm and then come back the zombie appears once more. Invisible zombie in the image below, notice that I can hear it by the subtitle line "zombie groans".
![image](https://user-images.githubusercontent.com/78611315/107063529-4047f380-67b9-11eb-9bf8-1652b75ff3aa.png)

**Comment by takinotomo** [comment]
Hey guys just want to contribute to the matter~

It's just speculation but could it possibly be related to the "Chunk Updates" option under Performance Settings?
The bug seems related to chunk loading, and I initially speculated it to be due to the Lazy Chunk Loading option but some users seem to encounter this on multiplayer too, so.
Can anybody who can reproduce this bug consistently test it with Chunk Updates set at 5?

btw @sp614x if you have trouble reproducing the bug try brutes in bastion remnants I think it is the easiest way, since the bug is kind of random and there are multiple of them in a bastion remnant.

Thanks a lot everyone. Peace.

**Comment by Lythael** [comment]
> It appears that this bug is related to the "smooth world" perfomance option, turning it ON makes the zombie in my iron farm disappear, if I turn it OFF and get far away from the farm and then come back the zombie appears once more. Invisible zombie in the image below, notice that I can hear it by the subtitle line "zombie groans".

I think you might be onto something.
I turned off Smooth World after reading your comment and to this day i haven't noticed any Invisible Drowned or Bastion Piglins in newly loaded chunks so far while usually it was 99% chance of meeting at least one.

**Comment by Horos402** [comment]
> Also they're not just invisible, they're invincible to melee attacks. This is not just a visual glitch.

I tried shooting them with projectile weapons like bow, crossbow, trident and I can kill them.

**Comment by Lythael** [comment]
> > Also they're not just invisible, they're invincible to melee attacks. This is not just a visual glitch.
> 
> I tried shooting them with projectile weapons like bow, crossbow, trident and I can kill them.

Invincible to MELEE ATTACKS. Ranged attacks can kill them just fine but melee attacks don't register at all.

**Comment by thealagator** [comment]
Trying to contribute as well here. My experience playing in multiplayer server / realms this already happened in pretty much every hostile mobs as far as I can remember, including skeletons, creepers, zombies, wither skeletons, piglins, zombified piglins, hoglins, and the baby variants. I already lost count how many times I died because of this.

> Invincible to MELEE ATTACKS. Ranged attacks including Trident throw can kill them just fine but melee attacks don't register at all.
I remember specifically killing a few glitched mobs by randomly swinging my swords, so I'm guessing this is purely visual (gotta admit it's pretty hard to even get a hit once, I got a clean hit by running into a corner and start spamming clicks). And if I remember correctly even when hitting one of them even with fire aspect sword the fire animation isn't even visible on them though I can still hear them taking fire damage. 

I use only some datapacks from vanillatweaks and I mostly play with no shaders. I don't tweak any setting other than what's already installed iirc. 

Hopefully this gets fixed soon.

[optionsof.txt](https://github.com/sp614x/optifine/files/5975216/optionsof.txt)
[options.txt](https://github.com/sp614x/optifine/files/5975214/options.txt)

**Comment by iobiarg** [comment]
I'm having the same issue, but it affects all entities, both modded and not modded.
Villagers appear some with normal body and white head, others in black with almost all the shaders I've tried, but with sildurs, and there they are transparent.
Modded animals have weird black parts.
Humanoid enemies the same.

I was running on a msi gtx 560 ti, and i though "okey, this thing is old AF, maybe if I get a newer this will be fixed", so I bought a used Evga 1060 3gb, but for my surprise the glitch is still there, just with more fps... -.-°

Next to me, there is a super-duper modern pc from my bf, with a graphic card that cost the double than my full computer, and I copied every single mod, shader, forge version, optifine, and he is not having the issue... so I feel like killing myself since I installed Mc there just to try this issue, because he doesn't like the game....

So, is there a way to use a fricking shaderpack but with entities working or I have to give up on the idea...?

**Comment by sp614x** [comment]
The invisible mobs are now 100% reproducible
thanks to @Zombie (discord) for providing the exact steps

**Comment by sp614x** [comment]
There is a specific combination of factors that triggers the bug
1. Smooth World ON - this seems to be the main trigger
2. Render distance between 8 and 16
3. Entity Distance 100%
4. Move along the chunk diagonals for 7 or 8 chunks, then come back

- render distance below 8 or above 16 doesn't have the problem
- moving less than 8 chunks doesn't have the problem
- moving more than 9 chunks only rarely has the problem
- entity view distance below 100% and above 150% doesn't have the problem

And the most strange:
- moving along the world axes doesn't trigger it for any distance, only when moving along the chunk diagonals

The simple solution for now is to disable Smooth World

**Comment by sp614x** [comment]
The client and server mobs were getting out of sync in a very specific distance from the player when smooth world was enabled

The entity behavior have been changed in 1.15.x, but OF 1.15.x is not affected as there the client entities know that they are persistent
In 1.16.x the client entities don't know that they are persistent and the desync allows them to despawn.
1.14.x and older shouldn't be affected.

This specific bug is also present in 1.15.x, but can't produce invisible entities because the client entities know they shouldn't despawn.
The same bug is not present in 1.14.x, the logic there is correct.

Reports for older versions may be a different bug, mod conflicts or even a vanilla bug.

**Comment by sp614x** [comment]
Fixed in 1.16.5_HD_U_G7_pre9

**Comment by Lythael** [comment]
> Fixed in 1.16.5_HD_U_G7_pre9

Nice thanks. Hopefully this issue is gone for good now.

**Comment by iobiarg** [comment]
at least for me the issue is still there with the new preview and changing to the settings you mention... T_T 
i get this issue with most of the shadders but with two as i posted on discord.. let me know if this information is of any help

**Comment by takinotomo** [comment]
IT IS ACTUALLY GETTING FIXED HALLELUJAH

**Comment by NoMoreStars** [comment]
Just thought I'd mention that I'm still experiencing problems with invisible mobs in 1.16.5_HD_U_G7_pre10 when I encountered an invisible Zombie Villager the other day. Strangely enough, I was able to see a small bit of a texture on it (it may have been wearing some modded armor?) and was able to damage it, but I think that bug is still there as of writing.

**Comment by iobiarg** [comment]
> Just thought I'd mention that I'm still experiencing problems with invisible mobs in 1.16.5_HD_U_G7_pre10 when I encountered an invisible Zombie Villager the other day. Strangely enough, I was able to see a small bit of a texture on it (it may have been wearing some modded armor?) and was able to damage it, but I think that bug is still there as of writing. Here's also an example of a villager with a broken texture that I'm assuming is related to this issue.
> 
> ![2021-02-18_09 07 45](https://user-images.githubusercontent.com/38673759/108377189-63b36b00-71c9-11eb-8f00-b39f6c21594e.png)

I have the same issue, but Xavi from MakeUpUltraFast, made a version of his shader that fix this, (but some of the nether mobs still have problems)

**Comment by NoMoreStars** [comment]
> 
> 
> 
> 
> I have the same issue, but Xavi from MakeUpUltraFast, made a version of his shader that fix this, (but some of the nether mobs still have problems)

Thought I'd give the MakeUp shaders a try, and while it does fix invisible textures, the textures are still broken, just no longer transparent. But I think this at least confirms that the issue of broken textures on mobs isn't exclusive to any one shaderpack.

![2021-02-18_12 17 21](https://user-images.githubusercontent.com/38673759/108402767-910d1280-71e3-11eb-8eeb-d261b9e4fcfb.png)

**Comment by iobiarg** [comment]
Check the shader config, there should be a option on the bottom left... if not talk to Xavi, 
![2021-02-18_15 30 19](https://user-images.githubusercontent.com/36169671/108404163-4e0c6880-71fe-11eb-8dff-6c84c43b07f9.png)

**Comment by iobiarg** [comment]
There are other 2 shaders that not give black entities, but one gives OpenGL error all the time (vanilla plus), and the other at least in my case that I have an old computer, it kills the fps (NV Shader),

Anyhow, if you talk to Xavi he is usually willing to help everyone (you can tell by the comments)

**Comment by thatdude69420** [comment]
Hi guys. I had a similar problem where some entities would be invisible. And you would see two tile like features just hovering in the air in place of the entity. I am however able to attack and kill the entity.

The solution that worked for me was to simply turn off fast render in the performance settings. video settings/performance/fast render

But note that I have my mipmaps and antialiasing off. I also have my render distance to 2 chunks. My chunk update is 1. Smooth fps/world can either be on or off. I have a low end pc. I can't speak for higher end pc's or high settings. I do not have any shaders on. I have resource packs installed (along with pixel luminance). I also have about a 100 mods installed. So yea...

I hope this can help someone.

**Comment by Sir-Satan** [comment]
Has there been a fix for this yet? me and my wife play mc together a lot and using Optifine really helps with fps, i play with her on LAN servers exclusively and seem to have mobs becoming invisible, players becoming invisible if they spawn outside of the loaded chunks (if someone spawns in and the player isn't in a chunk they load they become invisible for some reason) and that in order to fix this you can relog into the world or use a mod to pick up the player that can't see you(idk why this works but it does, i mainly use tinkers piggy back to do this).

**Comment by SeaOfPixels** [comment]
I haven't run into this bug for a long time I just assumed it was fixed. Try updating your OptiFine if you haven't.

**Comment by Sir-Satan** [comment]
> I haven't run into this bug for a long time I just assumed it was fixed. Try updating your OptiFine if you haven't.

it's the latest version of 1.16.5 and hasn't been updated in a year

**Comment by NatsuiroShiroi** [comment]
Hello guys, i put in Google "optifine invisible entities" and this thread came up, so I will comment on my problem here seeing that there are some who also suffered from it in past versions. With the optifine mod installed for 1.18.2 (version OptiFine_1.18.2_HD_U_H7) and practically all the performance improvements activated in the graphic options I get very good fps, and since there is no other mod for Forge as good (there is for Fabric ) I have come here to find a solution.

First I will explain the context in which the problem is developed:

I'm playing with the MineColonies mod and a few others that don't affect the graphics. The colony expands in chunks as I build new buildings, and once I got past 12 chunks of render distance and simulation distance, two different options, the entities that were outside of there could become invisible and you couldn't touch them even if you listen to them. If I moved away from some chunks and got closer to them, they would appear again, and that's quite annoying if I want to go from one point to another to work with npcs and they don't appear.

But the worst came today when i had several villagers with books of all kinds stored in a room that had open. Several pillagers appeared and I killed the one with the banner and a raid began, which in principle is not difficult to kill. What happened? Of the first five pillagers, only four appeared, and the last one remained invisible because it sure spawned a chunk more than it should have, right on the edge. I checked the minimap, all over town, underground and nothing. I went to where the villagers were, I went out to the main menu, I went back in and surprise surprise, 7 pillagers from the next raid were waiting for me in my face next to the villagers. Well, I killed them thinking it was a temporary graphical bug, but the raid didn't send me any more waves and I decided to go back to the main menu and go back in. Surprise surprise again, a ravager and 8 pillagers appeared right in front of me, they killed me and they killed a villager and a colony guard, all because they were invisible, and when I say they were invisible it's not that they could be beaten, it's that they still existed in the game and were moving as expected, but the player is not renderering them and it is impossible to locate them.

When the MineColonies npcs started to become invisible I thought it was because of that mod with some graphical glitch, which wouldn't make much sense either, but now I know it's because of Optifine, either because of the version itself or because of the mod itself, because it affected a vanilla part of the game without any mod content.

I don't know if there is any solution for the problem, some graphic option that I have to modify so that this stops happening. The mod optimizes the fps well, but I will have to stop using it if it starts to give these serious problems at a general level on the server I host.

If the person in charge of the mod or any other user can help me with this problem I would greatly appreciate it.
